src/Matrix.cpp

```diff
@@ -283,6 +325,21 @@ FP_PRECISION* Matrix::getA() {
 
 
 /**
+ * @brief Get the A component (excluding the diagonal) of the CSR form of the
+ *        matrix object.
+ * @return A pointer to the A component (excluding the diagonal) of the CSR
+ *         form matrix object.
+ */
+FP_PRECISION* Matrix::getAD() {
+
+  if (_modified)
```

Comment:
perhaps just add a comment preceding this block to describe what this actually means - *e.g.*, what does `_modified` represent?